accepted/00104-enhanced-ansible-support.md

```diff
@@ -0,0 +1,296 @@
+5) Make changes to the `RecurringActionJob` taskomatic job to support the new recurring action type.
+6) Implement API endpoints to support the managing of recurrent Ansible playbook execution.
+
+This newly created action type would be a minion only recurring action that would only be configurable from Ansible control nodes.
+
+As for the entry point to the new UI I see two options:
+
+1) On `System` > `Ansible` > `Playbooks` UI create a `Schedule Recurring` button that will link to `System` > `Recurring Actions` > `Create`. Here we will add the `Ansible Playbook` option to the dropdown to select the recurring action type. The action specific UI would then generate based on the type selected (like we do for existing types).
+
+2) Create a new `Recurring (Playbooks)` tab in `System` > `Ansible` that would allow the creation of Ansible Playbook recurring actions without having to select the action type first. Viewing created action details as well as editing/deleting existing actions would still have to be done from the `System` > `Recurring Actions` tab.
+
+## Set default playbook and inventory paths when adding a new control node in Uyuni
+
+This will be straightforward. Function to add playbooks/inventories already exists. We'll just have to call them with our wanted defaults when adding new control nodes.
+
+These defaults will be:
+
+Playbooks: `/etc/ansible/playbooks/`
+
+Inventory: `/etc/ansible/hosts`
+
+## Add option to filter by Ansible managed systems onboarded in Uyuni in the system list.
+
+We will add a new entity called `ansible_managed (Ansible Managed)` to the `rhnServerGroupType` and `rhnServerGroup` tables.
+To get the list of systems managed by an Ansible control node we'll scan all the inventories registered in `suseAnsiblePath`.
+
+There is currently a salt state called `ansible.targets inventory=<some_inventory>` available that returns inventory data. The list of systems can be parsed from that data (this mechanism was already part of the original Ansible implementation).
+
+However there are two problems that need two be solved here:
+1) We need a way to automate collecting Ansible managed systems to make sure the list is up to date.
+
+    This can be achieved by adding a `inotify beacon` on every control node. This beacon would contain the list of inventories registered on Uyuni and trigger an event to update the list of systems whenever one of the inventory files receives an update.
+
+    See example beacon below:
+    ```
+    beacons:
+      inotify:
+        - files:
+            /etc/ansible/hosts:
+              mask:
+                - modify
+            /another/inventory/file:
+              mask:
+               - modify
+        - interval: 5
+        - disable_during_state_run: True
+    ```
+
+    This beacon would be updated whenever a inventory file is added/removed on Uyuni.
+
+    Along with the beacon we'll have to add a new `reactor` to the salt master that will react to events send by the beacons.
```

Comment:
What would this reactor do? Shouldn't we just process the event? Save it on the database and let the salt event processor handle it?

Reply:
Yeah, I think it is better to use the `SaltReactor.java` here and not the Salt Master Reactors.